### ISSUE TITLE
[css-ruby] Align ruby with line head or line end

### ISSUE THREAD

**Issue by xfq** [w3c/csswg-drafts#4857]
In [clreq § 3.3.4 Positioning of Romanized Interlinear Annotations](https://w3c.github.io/clreq/#h-basic-requirements):

> If a Romanized annotation is longer than its base text and is at the line head or end, both the annotation and the base text can be aligned to the beginning of the line head or end.

This doesn't seem to be possible with the current selectors and ruby styling in CSS, so we probably need new `ruby-align` value(s) or new CSS selector(s).

**Comment by upsuper** [comment]
I guess either we need to have `ruby-align` accept multiple values, or have a new property for it.

**Comment by xfq** [comment]
There are similar requirements in Japanese:

> When the ruby string protrudes from the base character string, and the base character string is at the start or the end of the line, whether the base character string or the ruby string should be aligned with the line edge

See [simple-ruby](https://w3c.github.io/simple-ruby/#the-difficulties-of-ruby-processing).

**Comment by xfq** [comment]
See also [simple-ruby § 1.2](https://w3c.github.io/simple-ruby/#matters-considered-by-the-placement-rules):

> Also, this document does not take a method to align the first or last character of the ruby base character string to the line head or the line end, by modifying the relative position of the ruby string and the ruby base character string when the ruby base character string is placed at the line head or the line end.

**Comment by kojiishi** [comment]
The method you described used to be used in Japanese too, and there are existing publications that use this method, but it is rarely used these days, and that simple-ruby does not take this option. It can align line edges better, but as a side effects, it makes spaces longer. Modern typography prioritizes to minimize expansions.

Is this method common in modern Chinese typography?

**Comment by frivoal** [comment]
I agree with @kojiishi about simple ruby. Unlike clreq, it says that you should *not* align both the base and annotation to the line edge, only the longer of the two.

The behavior discussed in the opening comment, as well as the behavior recommended in simple ruby, is currently allowed, but not required, by the css-ruby spec in https://drafts.csswg.org/css-ruby-1/#line-edge.

My suggestion would be to continue to leave it up to implementations for now (and implementations could chose to do this in Chinese but not Japanese if they want to), close this issue with no change, and if we really need control over it, we can later add a new property in ruby level 2, with an initial value of auto.

Defining the various variants of line-edge ruby alignment properly would need to give more details about how overhang works (and that too is currently up to the user agent), and probably needs to interact with optical alignment questions, and that seems like too much for level 1.

**Comment by xfq** [comment]
> Is this method common in modern Chinese typography?

I don't think it's common in modern Chinese typography, but it does exist. Other clreq editors/contributors also [agree](https://www.w3.org/2020/06/10-clreq-minutes.html#t07) with me.
